File: Logic Building Problems/2_multiplication_table.cpp
#include <iostream>
using namespace std;


void timesTable(int n) {
    for (int i=1; i<=10; i++) {
        cout << n << " * " << i << " = " << n*i << endl;
    }
}


void timesTableWithRecursiveApproach(int n, int i=1) {
    if (i == 11) 
        return;
    cout << n << " * " << i << " = " << n*i << endl;
    i += 1;
    timesTableWithRecursiveApproach(n, i);
}

int main() {
    int input_number=0;
    cin >> input_number;
    // timesTable(input_number);
    timesTableWithRecursiveApproach(input_number);
    return 0;
}

File: Logic_Building_Problems/2_multiplication_table.py
def times_table(number):
    for i in range(1, 11):
        print(number, "*", i, "=", number*i)
        
def times_table_with_recursive_method(number, i=1):
    if (i == 11):
        return
    print(number, "*", i, "=", number*i)
    i += 1
    times_table_with_recursive_method(number, i)
        

if __name__ == "__main__":
    input_number = int(input())
    # times_table(number=input_number)
    times_table_with_recursive_method(input_number)